webrtc/stun/stun_transport.nim

```diff
@@ -0,0 +1,93 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import tables
+import chronos, chronicles, bearssl
+import stun_connection, stun_message, ../udp_connection
+
+logScope:
+  topics = "webrtc stun stun_transport"
+
+type
+  Stun* = ref object
+    connections: Table[TransportAddress, StunConn]
+    pendingConn: AsyncQueue[StunConn]
+    readingLoop: Future[void]
+    conn: UdpConn
+    rng: ref HmacDrbgContext
+
+proc accept*(self: Stun): Future[StunConn] {.async: (raises: [CancelledError]).} =
+  ## Accept a Stun Connection
+  ##
+  var res: StunConn
+  while true:
+    res = await self.pendingConn.popFirst()
+    if res.closed != true: # The connection could be closed before being accepted
+      break
+  return res
+
+proc connect*(
+    self: Stun,
+    raddr: TransportAddress
+  ): Future[StunConn] {.async: (raises: []).} =
+  ## Connect to a remote address, creating a Stun Connection
+  ##
+  var res = self.connections.getOrDefault(raddr)
```

Comment:
what about https://nim-lang.org/docs/tables.html#withValue.t%2CTable%5BA%2CB%5D%2CA%2Cuntyped%2Cuntyped%2Cuntyped

Reply:
I forgot withValue was in Table... My bad, it's clearly better :+1: 
I replaced it!